src/main/kotlin/org/arctgkolbasy/bot/handler/CurrentUserHandler.kt

```diff
@@ -15,13 +15,19 @@ class CurrentUserHandler(
     val currentUserHolder: ThreadLocal<User?>,
     val commands: List<SecuredCommand>,
 ) : Handler {
-    override fun checkUpdate(update: Update): Boolean = update.message?.from != null
+    override fun checkUpdate(update: Update): Boolean =
+        update.message?.from != null || update.callbackQuery?.from != null
 
     override fun handleUpdate(bot: Bot, update: Update) {
-        val user = userService.getOrCreateUser(update.message!!.from!!)
+        val getUser = let { update.message?.from } ?: update.callbackQuery?.from
+        val user = userService.getOrCreateUser(getUser!!)
         currentUserHolder.set(user)
-        val command = commands.singleOrNull {
-            update.message?.text?.startsWith("/${it.getCommandName()}") == true
+        val command = let {
+            commands.singleOrNull {
+                update.message?.text?.startsWith("/${it.getCommandName()}") == true
+            }
+        } ?: commands.singleOrNull {
+            update.callbackQuery?.data?.startsWith(it.getCommandName()) == true
         }
```

Comment:
во-первых, здесь не нужен let. во-вторых, можно использовать "или"
```suggestion
        val command = commands.singleOrNull {
            update.message?.text?.startsWith("/${it.getCommandName()}") == true 
                || update.callbackQuery?.data?.startsWith(it.getCommandName()) == true
        }
```